include/VulkanInfoProvider.hpp

```diff
@@ -3,19 +3,20 @@
 #include <lug/Config.hpp>
 #include <lug/Graphics/Renderer.hpp>
 #include <lug/Graphics/Vulkan/Renderer.hpp>
+#include <lug/Graphics/Vulkan/API/RTTI/Enum.hpp>
 
 #if defined(LUG_SYSTEM_WINDOWS)
-    #pragma warning(push)
-    #pragma warning(disable : 4003)
+#pragma warning(push)
+#pragma warning(disable : 4003)
 #endif
 #include <json/json.hpp>
 #if defined(LUG_SYSTEM_WINDOWS)
-    #pragma warning(pop)
+#pragma warning(pop)
 #endif
 
 class VulkanInfoProvider {
 public:
-    VulkanInfoProvider(lug::Graphics::Vulkan::InstanceInfo& instanceInfo);
+    VulkanInfoProvider(lug::Graphics::Vulkan::PhysicalDeviceInfo *physicalDeviceInfo);
```

Comment:
You always need the `PhysicalDeviceInfo`, you don't want `nullptr`. So use a reference instead of a pointer.